misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -15,24 +15,283 @@
+     *
+     * @param section the format section
+     * @param key     the format option
+     * @return the default integer value
+     */
+    public static int getDefaultInt(FormatSection section, String key) {
+        return Integer.parseInt(getDefaultString(section, key));
+    }
+
+    /**
+     * Retrieves the default string value for the format option in the given format section.
+     *
+     * @param section the format section
+     * @param key     the format option
+     * @return the default string value
+     */
+    public static String getDefaultString(FormatSection section, String key) {
+        return DEFAULTS.getString(section.getStringValue() + "." + key);
+    }
+
+    /**
+     * Loads the format section in the Ballerina.toml .
+     *
+     * @param manifest the package manifest
+     * @return the format section
+     */
+    public static Object loadFormatSection(PackageManifest manifest) {
+        return manifest.getValue(FORMAT);
+    }
+
+    /**
+     * Retrieves the formatting file path.
+     *
+     * @param formatSection the format section loaded from Ballerina.toml
+     * @param root          the root path of the project
+     * @return the formatting file path
+     */
+    public static Optional<String> getFormattingFilePath(Object formatSection, String root) {
+        if (formatSection != null) {
+            Object path = ((Map<String, Object>) formatSection).get(FORMAT_FILE_FIELD);
+            if (path != null) {
+                String str = path.toString();
+                if (str.endsWith(FORMAT_OPTION_FILE_EXT)) {
+                    return Optional.of(str);
+                }
+                return Optional.of(Path.of(str, DEFAULT_FORMAT_OPTION_FILE).toString());
+            }
+        }
+
+        Path defaultFile = Path.of(root, DEFAULT_FORMAT_OPTION_FILE);
+        return Files.exists(defaultFile) ? Optional.of(defaultFile.toString()) : Optional.empty();
+    }
+
+    /**
+     * Retrieves the formatting configurations.
+     *
+     * @param root the root path of the project
+     * @param configurationFilePath the path of configuration file
+     * @return the formatting configurations
+     * @throws FormatterException if the file resolution fails
+     */
+    public static Map<String, Object> getFormattingConfigurations(Path root, String configurationFilePath)
+            throws FormatterException {
+        String content;
+        Optional<Path> path = convertConfigurationPath(configurationFilePath);
+        Optional<Path> absPath = path.isEmpty() || path.get().isAbsolute() || !root.isAbsolute() ? path :
+                Optional.of(root.resolve(configurationFilePath));
+        if (isLocalFile(absPath)) {
+            try {
+                content = Files.readString(absPath.get(), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to retrieve local formatting configuration file");
```

Comment:
It is not empty here. Because in the `isLocalFile()` we return false if the path empty.
https://github.com/poorna2152/ballerina-lang/blob/formatter/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java#L202